app/src/main/java/com/epfl/drawyourpath/challenge/DailyGoalViewAdapter.kt

```diff
@@ -1,24 +1,33 @@
 package com.epfl.drawyourpath.challenge
 
-import android.text.Editable
-import android.text.TextWatcher
+import android.content.Context
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.view.inputmethod.EditorInfo
+import android.view.inputmethod.InputMethodManager
 import android.widget.EditText
 import android.widget.ProgressBar
 import android.widget.TextView
 import androidx.recyclerview.widget.RecyclerView
 import com.epfl.drawyourpath.R
+import com.epfl.drawyourpath.userProfile.dailygoal.DailyGoal
 import java.lang.Integer.min
+import java.util.concurrent.CompletableFuture
 
 
 /**
  * used in a recycler view to display the [DailyGoal]
  */
-class DailyGoalViewAdapter(private val dailyGoal: DailyGoal) :
+class DailyGoalViewAdapter(
+    private val setDistanceGoal: (distance: Double) -> CompletableFuture<Unit>,
+    private val setTimeGoal: (time: Double) -> CompletableFuture<Unit>,

```

Comment:
Maybe you can rename it to setActivityTimeGoal fo consistency with the rest of the app.